File: frontend/permission-admin/src/api/ssoLoginLog.ts
import { request } from '../utils/request'
import type { ApiResult, PagedResult, PageQuery } from './types'
import type { SsoProviderType } from './ssoProvider'

export const SsoLoginResult = {
  Success: 0,
  Failed: 1,
  UserDisabled: 2,
  TenantDisabled: 3,
  BindingFailed: 4,
  AutoCreateFailed: 5,
} as const

export type SsoLoginResult = (typeof SsoLoginResult)[keyof typeof SsoLoginResult]

export interface SsoLoginLogQuery extends PageQuery {
  providerCode?: string
  providerType?: SsoProviderType
  loginResult?: SsoLoginResult
  startAt?: string
  endAt?: string
}

export interface SsoLoginLogItem {
  id: string
  tenantId: string
  providerCode: string
  providerName: string
  providerType: SsoProviderType
  externalUserId?: string
  externalUserName?: string
  localUserId?: string
  localUserName?: string
  loginResult: SsoLoginResult
  failureReason?: string
  ipAddress?: string
  userAgent?: string
  traceId?: string
  createdAt: string
}

export function getSsoLoginLogs(params: SsoLoginLogQuery) {
  return request
    .get<ApiResult<PagedResult<SsoLoginLogItem>>>('/api/sso/login-logs', { params })
    .then((res) => res.data.data)
}

export function getSsoLoginLog(id: string) {
  return request.get<ApiResult<SsoLoginLogItem>>(`/api/sso/login-logs/${id}`).then((res) => res.data.data)
}

File: backend/PermissionSystem.Domain/Entities/SsoLoginLog.cs
using PermissionSystem.Domain.Common;
using PermissionSystem.Domain.Enums;

namespace PermissionSystem.Domain.Entities;

public sealed class SsoLoginLog : BaseEntity
{
    public string ProviderCode { get; set; } = string.Empty;

    public string ProviderName { get; set; } = string.Empty;

    public SsoProviderType ProviderType { get; set; } = SsoProviderType.Oidc;

    public string? ExternalUserId { get; set; }

    public string? ExternalUserName { get; set; }

    public Guid? LocalUserId { get; set; }

    public string? LocalUserName { get; set; }

    public SsoLoginResult LoginResult { get; set; } = SsoLoginResult.Failed;

    public string? FailureReason { get; set; }

    public string? IpAddress { get; set; }

    public string? UserAgent { get; set; }

    public string? TraceId { get; set; }
}
